Branch-SDK/src/io/branch/referral/Branch.java

```diff
@@ -2346,14 +2367,14 @@ private void registerAppInit(BranchReferralInitListener
         if (isGAParamsFetchInProgress_) {
             request.addProcessWaitLock(ServerRequest.PROCESS_WAIT_LOCK.GAID_FETCH_WAIT_LOCK);
         }
-        if (intentState_ != INTENT_STATE.READY) {
+        if (intentState_ != INTENT_STATE.READY && !isForceSessionEnabled()) {
```

Comment:
We still run the risk of stale intent without this, but I guess this is cleaner.  We should make it clear that they should _only_ do this force session call from the observer for attribution purposes. If we do this for actual deep linking, it may be an issue.

Reply:
Can we have this commented in the above block? This is helpful context that I'd wanna make sure is documented.

Reply:
👍